### ISSUE TITLE
Assertion failed when disabling ref counts and returning a string literal

### ISSUE THREAD

**Issue by harold-b** [svaarala/duktape#1432]
I'm getting "**assertion failed: !DUK_TVAL_IS_HEAP_ALLOCATED(tv)**" in v2.0.2 (though I also tried in v2.0.0 with the same result) when disabling `DUK_USE_REFERENCE_COUNTING` and returning a string literal from a function. It doesn't do so with any other value type, or if you assign the string to a var.

The assert occurs this [line on duk_executor.c](https://github.com/svaarala/duktape/blob/master/src-input/duk_js_executor.c#L3943).

Enabling ref counting fixes it.

C repro:
``` c
int main( int argc, const char* argv[] )
{
    const char js[] = "function() { return '' }";
    duk_context* cx = duk_create_heap_default();
  
    duk_push_string( cx, js );
    duk_push_string( cx, "main.js" );
    duk_pcompile( cx, DUK_COMPILE_FUNCTION );
    duk_pcall( cx, 0 );
    
    return 0;
}
```

With config:
``` bash
-DDUK_USE_ASSERTIONS                \
-UDUK_USE_REFERENCE_COUNTING
```

Call Stack:
```
 	DukBug.exe!duk_default_fatal_handler(void * udata, const char * msg) Line 126	C
>	DukBug.exe!duk__js_execute_bytecode_inner(duk_hthread * entry_thread, unsigned int entry_callstack_top) Line 3917	C
 	DukBug.exe!duk_js_execute_bytecode(duk_hthread * exec_thr) Line 2362	C
 	DukBug.exe!duk__handle_call_inner(duk_hthread * thr, int num_stack_args, unsigned int call_flags, int idx_func) Line 1570	C
 	DukBug.exe!duk_handle_call_protected(duk_hthread * thr, int num_stack_args, unsigned int call_flags) Line 1051	C
 	DukBug.exe!duk_pcall(duk_hthread * ctx, int nargs) Line 134	C
 	DukBug.exe!main(int argc, const char * * argv) Line 17	C
```

**Comment by svaarala** [comment]
That assert is incorrect when reference counting is disabled: because no refcount update is done, RETCONSTN is used for all return values. The actual behavior is correct though, so the assert doesn't indicate a bug. I fixed the assert in #1427 (to be cleaned up and merged).

**Comment by svaarala** [comment]
Hmm, but I think I'd better cherry pick the fix for 2.0.3.

**Comment by harold-b** [comment]
> The actual behavior is correct though, so the assert doesn't indicate a bug.

I figured it wouldn't, but thought I'd let you know anyway :wink:

**Comment by svaarala** [comment]
Appreciate that, thanks :) Assert problems are bugs to be fixed for sure.

**Comment by svaarala** [comment]
Ok, the fix from #1433 is merged to v2.0-maintenance too - in case you want to check that it got fixed for you too: https://github.com/svaarala/duktape/tree/v2.0-maintenance.

**Comment by harold-b** [comment]
Yup, working great now over here too.
Thanks for the quick fix. :)
